invokeai/app/util/save_thumbnail.py

```diff
@@ -0,0 +1,28 @@
+import os
+from PIL import Image
+
+
+def save_thumbnail(
+    image: Image.Image,
+    filename: str,
+    path: str,
+    size: int = 256,
+) -> str:
+    """
+    Saves a thumbnail of an image, returning its path.
+    """
+    base_filename = os.path.splitext(filename)[0]
+    thumbnail_path = os.path.join(path, base_filename + ".webp")
```

Comment:
Why are these webp?

Reply:
Small file size. Loading a ton of jpeg / png files in the gallery as thumbnails can get resource intensive. So we shifted to using webp's for the thumbnails in the gallery.

Reply:
For reference, 512x512 png result vs 256x256 thumbnail:

```bash
520K    fea6cf48-ad19-4d31-a583-e36879645773_449210a3-5243-4b6c-abf2-c65458a56220_1680611408.png
20K     thumbnails/fea6cf48-ad19-4d31-a583-e36879645773_449210a3-5243-4b6c-abf2-c65458a56220_1680611408.webp
```